pkg/port-forward.go

```diff
@@ -36,12 +36,22 @@ func (w WebsocketPortForward) Write(p []byte) (n int, err error) {
 	return len(p), err
 }
 func (w WebsocketPortForward) Read(p []byte) (n int, err error) {
-	_, msg, err := w.ws.ReadMessage()
-	if err != nil {
-		return 0, err
-	}
+	for {
+		msgType, msg, err := w.ws.ReadMessage()
+		if err != nil {
+			return 0, err
+		}
+
+		if msgType != websocket.CloseMessage {
+			return 0, io.EOF
+		}
 
-	return copy(p, msg), err
+		if msgType != websocket.BinaryMessage {
```

Comment:
I don't get how it can pass both condition, or maybe I'm missing something:
* if msgType == CloseMessage, it will continue (because != BinaryMessage)
* if msgType == BinaryMessage, it will return error (because != CloseMessage)
* if msgType == any other type, it will return error (because != CloseMessage)
When does it reach the `return copy(p, msg)` code ? :thinking:

Reply:
Yeah, saw the shell.go below which is OK :see_no_evil: